aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -331,34 +300,43 @@ module post_mint_reveal_nft::minting {
             let reveal_config = borrow_global_mut<RevealConfig>(@post_mint_reveal_nft);
             reveal_config.reveal_time = reveal_time;
         } else {
-            let resource_account = create_signer_with_capability(&nft_mint_config.signer_cap);
-            move_to(&resource_account, RevealConfig {
+            move_to(admin, RevealConfig {
                 reveal_time,
             });
         };
     }
 
-    /// Add user addresses to the whitelist.
+    /// Set up different stages of the whitelist config.
+    /// Most of the times, NFT projects will only have one-stage of whitelist.
+    /// In that case, the `curr_stage` being passed in should be 0.
+    /// If there are multiple stages of whitelist, the `curr_stage` should be the
+    /// current stage of the whitelist (starting from 0).
+    /// For example, if there are 3 stages of whitelist,
+    /// we need to call this function three times, with curr_stage being 0, 1, 2 respectively.
+    public entry fun set_up_whitelist_stage(
+        admin: &signer,
+        whitelist_start_time: u64,
+        whitelist_end_time: u64,
+        whitelist_price: u64,
+        whitelist_stage: u64,
+    ) {
+        assert!(signer::address_of(admin) == @post_mint_reveal_nft, error::permission_denied(ENOT_AUTHORIZED));
```

Comment:
Why is this check necessary? If this is called with a signer other than @post_mint_reveal_nft, the whitelist config on that account would be used instead, which effectively seems like the right behavior

Reply:
was trying to limit access control to only the minting module, but can remove this check too

Reply:
Can you add a comment to document this requirement? May be add a line at the top to say all admin functions are restricted to the deployer account as well.